plugins/modules/sysctl.py

```diff
@@ -187,6 +231,13 @@ def process(self):
 
         # Do the work
         if not self.module.check_mode:
+            # Set file permissions if there are differences.
+            # - Ansible 2.9.x does not support 'path' like load_file_common_arguments (params, path=PATH).
+            #   so set 'sysctl_file' as 'path' in module.params.
+            self.module.params['path'] = self.sysctl_file
+            file_args = self.module.load_file_common_arguments(self.module.params)
+            self.changed = self.module.set_fs_attributes_if_different(file_args, self.changed)
```

Comment:
`module.set_fs_attributes_if_different()` supports `check_mode` on its own. This call should take place outside of the `if not self.module.check_mode:` branch, so when playing the module in `check_mode`, if there are differences to apply in file properties, the `set_fs_attributes_if_different()` is aware of them and will report the change, and will report the *same* change, with `check_mode=true` or `false`.
This is currently not the case: with the call to `set_fs_attributes_if_different()` skipped if `check_mode=true`, the module is unable to compute/return changes in file `mode` or `owner`.

Reply:
> module.set_fs_attributes_if_different() supports check_mode on its own.

Thank you for your advice! I have understood that set_fs_attributes_if_different already have check logic it self using `check_file_absent_if_check_mode()`

* https://github.com/ansible/ansible/blob/96d2f79e4c349a7cbff4e5aaf11c39bac75a474e/lib/ansible/module_utils/basic.py#L1177-L1178

So I'll move this logic outside of `if not self.module.check_mode:` block.

Reply:
done :)